configs/components/_base-ruby.rb

```diff
@@ -50,7 +50,7 @@
   pkg.environment 'optflags', settings[:cflags] + ' -O3'
 elsif platform.is_macos?
   pkg.environment 'optflags', settings[:cflags]
-  if platform.is_cross_compiled?
+  if platform.architecture == 'arm64'
     # Pin to an older version of ruby@2.5. This can be removed once we're no longer cross-compiling
```

Comment:
I think we should be able to remove this conditional in pinning to ruby 2.5?

Reply:
But we still want to cross-compile for osx-11 and osx-12

Reply:
We can exclude this change after rebasing, see https://github.com/puppetlabs/puppet-runtime/pull/680/files#r1253820661. 

Also @tvpartytonight is right that Ruby 2.5 is no longer needed (it was used in Puppet 6 which is EOL). If there isn't a PA ticket filed already about removing 6.x code from the runtime, can you file one and we can delete that code later?